django/pyrog/api/serializers/basic.py

```diff
@@ -9,15 +13,55 @@ class Meta:
         fields = "__all__"
 
 
-class ResourceSerializer(serializers.ModelSerializer):
+class CredentialSerializer(serializers.ModelSerializer):
+    available_owners = serializers.SerializerMethodField()
+
     class Meta:
-        model = models.Resource
+        model = models.Credential
         fields = "__all__"
 
+    def get_available_owners(self, obj) -> List[str]:
```

Comment:
where/how is this called ?

Reply:
It's part of the `available_owners` SerializerMethodField: https://www.django-rest-framework.org/api-guide/fields/#serializermethodfield 

You can declare a SerializerMethodField and name it then it will be automatically associated with the method named `get_{field_name}`.

Reply:
ok thanks. This means this is called everytime we GET a Credential right? I'm a little bit worried of performance if we have to init a db connection and get the list of owners everytime, but at the same time I find it simpler than before because the owner list was only refreshed when the credential was created/updated.

Reply:
On perfs: when we first discussed this, we talked about passing a query field parameter (such as `/api/sources/<id>/?introspect=True` to actually introspect the DB and add the field `available_owners`.
Maybe we could implement this if there is perf issues

Reply:
yes but I don't think it's that expensive. We only fetch once a credential and then it is cached by the app.

Reply:
thats the spirit :) (should have added "later" but it was implied)

Reply:
agreed!